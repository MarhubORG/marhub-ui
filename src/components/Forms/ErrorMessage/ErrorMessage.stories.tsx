import * as React from 'react';
import { storiesOf } from '@storybook/react';
import { ThemeProvider } from 'styled-components';
import { mainTheme } from '../../../styles/themes';
import ErrorMessage from './ErrorMessage';

storiesOf('ErrorMessage', module).add('ErrorMessage Component', () => (
  <ThemeProvider theme={mainTheme}>
    <ErrorMessage message="Incorrect Password" />
  </ThemeProvider>
));
