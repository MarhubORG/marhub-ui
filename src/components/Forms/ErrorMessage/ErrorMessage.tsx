import React from 'react';
import styled from 'styled-components';

interface Props {
  message?: string;
}

export default function ErrorMessage(props: Props): JSX.Element | null {
  return props.message ? <Layout>{props.message}</Layout> : null;
}

const Layout = styled.div`
  margin-top: 1rem;
  padding-top: 0.5rem;
  display: flex;
  justify-content: center;
  height: 2rem;
  min-width: 18rem;
  max-width: 18rem;
  border-radius: 0.2rem;
  font-size: 0.9rem;
  padding-left: 1rem;
  padding-right: 1rem;
  color: red;
  border: 1px solid #edf1f7;
`;
