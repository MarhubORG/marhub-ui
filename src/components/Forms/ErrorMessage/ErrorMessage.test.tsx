import React from 'react';
import renderer from 'react-test-renderer';
import ErrorMessage from './ErrorMessage';

it('renders correctly', () => {
  const tree = renderer.create(<ErrorMessage message="message" />).toJSON();
  expect(tree).toMatchSnapshot();
});
