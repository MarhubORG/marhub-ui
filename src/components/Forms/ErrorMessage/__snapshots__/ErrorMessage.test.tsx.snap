// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c0 {
  margin-top: 1rem;
  padding-top: 0.5rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 2rem;
  min-width: 18rem;
  max-width: 18rem;
  border-radius: 0.2rem;
  font-size: 0.9rem;
  padding-left: 1rem;
  padding-right: 1rem;
  color: red;
  border: 1px solid #edf1f7;
}

<div
  className="c0"
>
  message
</div>
`;
